openedx/core/djangoapps/user_api/accounts/tests/test_views.py

```diff
@@ -1019,6 +1019,39 @@ class RetirementTestCase(TestCase):
     """
     Test case with a helper method for getting JWT headers
     """
+    def setup_states(self):
+        default_states = [
+            ('PENDING', 1, False, True),
+            ('LOCKING_ACCOUNT', 2, False, False),
+            ('LOCKING_COMPLETE', 3, False, False),
+            ('RETIRING_CREDENTIALS', 4, False, False),
+            ('CREDENTIALS_COMPLETE', 5, False, False),
+            ('RETIRING_ECOM', 6, False, False),
+            ('ECOM_COMPLETE', 7, False, False),
+            ('RETIRING_FORUMS', 8, False, False),
+            ('FORUMS_COMPLETE', 9, False, False),
+            ('RETIRING_EMAIL_LISTS', 10, False, False),
+            ('EMAIL_LISTS_COMPLETE', 11, False, False),
+            ('RETIRING_ENROLLMENTS', 12, False, False),
+            ('ENROLLMENTS_COMPLETE', 13, False, False),
+            ('RETIRING_NOTES', 14, False, False),
+            ('NOTES_COMPLETE', 15, False, False),
+            ('NOTIFYING_PARTNERS', 16, False, False),
+            ('PARTNERS_NOTIFIED', 17, False, False),
+            ('RETIRING_LMS', 18, False, False),
+            ('LMS_COMPLETE', 19, False, False),
+            ('ERRORED', 20, True, True),
+            ('ABORTED', 21, True, True),
+            ('COMPLETE', 22, True, True),
+        ]
```

Comment:
This was one of the open questions I'd asked in HipChat after discovering that we couldn't feasibly do it via settings. I'll poke you about a configuration option, because I think it will be a major pain point if we can't make it relatively easily configurable.

As for these states, they're just hard coded into the test so I'm not even sure what the use case would be for adding more. If anything there are more than necessary now. Happy to increment them by 10s just in case, though.

Reply:
Sure - we can discuss configuration options.

Sorry - I was talking about the eventual non-test states having number space for more states. I agree that the test states probably have more states than necessary. We've already been talking about possibly splitting up the final grab-bag endpoint into a few states in a single endpoint.